top-level: Add a warning on deprecated aliases

nicoonoclaste opened the pull request:
## Description of changes

- [x] Programatically apply `lib.warn` on all aliases in `top-level/aliases.nix`
- [x] Include the alias' target's `pname` in the warning message, if it's a derivation.


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

nicoonoclaste committed:
top-level: Add deprecation warning to aliases

Artturin commented:
Previous attempts of this have failed because of nix-env and this may too.

Artturin commented:
I'm not sure where the PRs are, but it has definitely been tried before, may not have even been a PR but a comment.

Here's one commit https://github.com/nixos/nixpkgs/commit/5cd6c650545dbfe1d040bfbd00ad7ecc2f4f4763

and the CI for this PR fails

nicoonoclaste committed:
aliases: Fallback to drv.name in `deprecationWarning`

Some derivations currently in `aliases.nix` visibly do not
have a `pname` attribute defined.

Artturin commented:
The warnings show in `nix search` and `nix-env` and that's one reason why we don't have this.
